File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Hungary.m
//
//  UIImage+Hungary.m
//  CountriesFlags
//

#import "Countries/UIImage+Hungary.h"

@implementation UIImage (Hungary)

+ (UIImage * _Nullable)flagHungaryWithSize:(CGSize)size {
    UIColor* colorTomato = [UIColor colorWithRed:206.0f/255.0f
                                           green:41.0f/255.0f
                                            blue:57.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorFairway = [UIColor colorWithRed:71.0f/255.0f
                                            green:112.0f/255.0f
                                             blue:80.0f/255.0f
                                            alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorTomato setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0, size.width, size.height / 3.0));
    
    [colorFairway setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0 * 2.0, size.width, size.height / 3.0));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIIMage+Hungary.swift
//
//  UIIMage+Hungary.swift
//  CountriesFlags
//

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://uk.wikipedia.org/wiki/Прапор_Угорщини "wikipedia"
     
     Generates the flag of Hungary.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Hungary.
     */
    internal static func flagHungary(with size: CGSize) -> UIImage? {
        let colorTomato = UIColor(red: 206.0/255.0,
                                  green: 41.0/255.0,
                                  blue: 57.0/255.0,
                                  alpha: 1.0)
        let colorWhite = UIColor.white
        let colorFairway = UIColor(red: 71.0/255.0,
                                   green: 112.0/255.0,
                                   blue: 80.0/255.0,
                                   alpha: 1.0)
        
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        
        colorTomato.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))
        
        colorFairway.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))
        
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return image
    }
}
